### PR TITLE
Plugin Rename

### PR DESCRIPTION
**Pull request by mmitche**
Renames various terms (slave where able).  Allows for SxS deployment with classic model.

### CODE REVIEW

**Comment by arroyc** [comment]
Thanks @mmitche , will look into it. One question. I thought you said you want to keep this in a seperate branch till we decide if we are going to publish it as agent or slave v2.

**Comment by mmitche** [comment]
i thought that's why i was merging this into arm-dev (and not into master)?

File: src/main/java/com/microsoft/azure/AzureVMCloudVerificationTask.java [hunk 1]
```diff
@@ -0,0 +1,306 @@
...
+                        LOGGER.log(Level.INFO, "AzureVMCloudVerificationTask: execute: parent cloud not found for {0} in {1}", 
+                            new Object[] { templateName, cloudName });
+                        toRemove.add(templateName);
+                        continue;
+                    }
+
+                    AzureVMAgentTemplate agentTemplate = cloud.getAzureAgentTemplate(templateName);
+                    // Template could have been removed since the last time we ran verification
+                    if (agentTemplate == null) {
+                        LOGGER.log(Level.INFO, "AzureVMCloudVerificationTask: execute: could not retrieve slave template named {0} in {1}",
```
Comment (clguimanMSFT) [inline src/main/java/com/microsoft/azure/AzureVMCloudVerificationTask.java]: missed one :)

File: src/main/java/com/microsoft/azure/AzureVMCloudVerificationTask.java [hunk 2]
```diff
@@ -67,130 +67,124 @@ public AzureVMCloudVerificationTask() {
...
-                    continue;
-                }
-                
-                // If already verified, skip
-                if (cloud.isConfigurationValid()) {
-                    LOGGER.log(Level.INFO, "AzureVMCloudVerificationTask: execute: subscription {0} already verifed", cloudName);
-                    // Update the count.
-                    cloud.setVirtualMachineCount(getVirtualMachineCount(cloud));
-                    continue;
+        if (cloudNames != null && !cloudNames.isEmpty()) {
```
Comment (clguimanMSFT) [inline src/main/java/com/microsoft/azure/AzureVMCloudVerificationTask.java]: It's a bit hard to follow - Did anything change here except for negating the if statement and moving the code inside?

Comment (mmitche) [inline src/main/java/com/microsoft/azure/AzureVMCloudVerificationTask.java]: Nope, that's all that changed.

File: src/main/java/com/microsoft/azure/util/AzureUtil.java [hunk 3]
```diff
@@ -307,15 +307,15 @@ public static boolean isValidTemplateName(String templateName) {
...
+    public static String getDeploymentName(String templateName, Date timestamp) {
         if (!isValidTemplateName(templateName)) {
             throw new IllegalArgumentException("Invalid template name");
         }
         
         Format formatter = new SimpleDateFormat(Constants.DEPLOYMENT_NAME_DATE_FORMAT);
         return String.format("%s%s", getShortenedTemplateName(templateName, Constants.USAGE_TYPE_DEPLOYMENT, 
             Constants.DEPLOYMENT_NAME_DATE_FORMAT.length(), 0), 
-                formatter.format(new Date(System.currentTimeMillis())));
+                formatter.format(timestamp));
```
Comment (clguimanMSFT) [inline src/main/java/com/microsoft/azure/util/AzureUtil.java]: Do you think ms precision timestamp is enough? What if two builds start at the same exact time? 
I know the chances are much smaller, but from the user perspective he wouldn't be able to figure out what went wrong. 
Can we use a UUID or the size of that is too large?

Comment (clguimanMSFT) [inline src/main/java/com/microsoft/azure/util/AzureUtil.java]: Got it. I guess it makes sense
